File: Sorting Algo/SelectionSort.cpp
#include<iostream>
using namespace std;
void SelectionSort(int arr[],int n)
{
    for(int i=0;i<n-1;i++)
    {
        int j=i,k=i;
        for(;j<n;j++)
        {
            if(arr[j]<arr[k]) k=j;
        }
        swap(arr[i],arr[k]);
    }

}
void Display(int *arr,int n)
{
    for(int i=0;i<n;i++) cout<<arr[i]<<" ";

    cout<<endl;
}
int main()
{
    int arr[] = {8,6,4,1,2,3,7};
    int len = sizeof(arr)/sizeof(arr[0]);
    Display(arr,len);
    SelectionSort(arr,len);
    Display(arr,len);

    return 0;
}

File: Sorting Algo/SelectionSort.py
import random

def randomIn(n):
    num = []
    i = 0
    while i<n:
        x = random.randint(1,10)
        if x not in num:
            num.append(x)
            i+=1
    return num

arr = randomIn(5)
print(arr)

def minIndex(arr,low,high):
    m = arr[low]
    mIN  = low
    for i in range(low+1,high):
        if(arr[i]<m):
            m = arr[i]
            mIN = i
    return mIN

def selectionSort(arr=[],n=1):
    for i in range(n):
        m = minIndex(arr,i,n)
        arr[m],arr[i]=arr[i],arr[m]

selectionSort(arr,5)
print(arr)